### DISCUSSION TITLE
Exclude Specific Files In Grep

### DISCUSSION THREAD

**Discussion by m4kman** [ibhagwan/fzf-lua#1280] [Q&A] [answered]
Hey, 

I just switched to fzf-lua from telescope in LazyVim, and I would like to know how can I exclude certain files while performing a live_grep or grep? 

From reading around a bit, I believe this can be done via the grep rg_opts command? Something like the following: 

`:lua require'fzf-lua'.grep({ rg_opts = "--column --line-number --no-heading --color=always --smart-case --max-columns=4096 -e --exclude FILENAME", })`

Is there a way I can specify a pattern here that matches certain file names? For example, if I wish to ignore package-lock files, how would I achieve this?

With telescope, this could be done via **%-lock.json**, etc.

Thank you for the help. Cheers.

**Comment by ibhagwan** [comment, marked as answer]
You can use `file_ignore_patterns` for exactly that (lua style patterns).

**Reply by m4kman** [reply]
That was much easier than I thought, thank you. 

Not sure if this is a bug, but it seems to be grepping errors from completely different directories in the grep. I've shared a screenshot for reference. 


![image](https://github.com/ibhagwan/fzf-lua/assets/30337413/6a33bde3-c164-49d3-b6a4-a96593296c3a)

**Reply by ibhagwan** [reply]
`file_ignore_patterns` runs within fzf-lua so it still has to process all results (I.e. “other” directories) in order to do the filtering.

The specific error however seems like a bug,  if you can make your terminal bigger so I can see the entire stack trace I’ll fix it.

**Reply by ibhagwan** [reply]
Also pls let me know what you set as `file_ignore_patterns`?

**Reply by m4kman** [reply]
EDIT: The error seems to have been caused by a typo in my file_ignore_patterns table. Fixing the typo removed the error. Sorry for the inconvenience.

**Reply by ibhagwan** [reply]
I still can't see the full stack trace, I need to see which lines are failing, close the previewer with `F4` pls.

> Seems like fzf-lua is looking for files in the wrong location

I doubt it, the error is due to the bug, doesn't mean the search is in the wrong location, add `debug=true` to the options and you'll see the ripgrep command fzf-lua ran.

**Reply by m4kman** [reply]
My apologies, I edited my reply without refreshing. 

This happened to be the consequence of a typo in the file_ignore_patterns table on my part. No issues to report. Sorry for any inconvenience, again!

**Reply by ibhagwan** [reply]
No worries, ty for the update @m4kman.

**Reply by m4kman** [reply]
My pleasure, thank you for the assistance.
